xls/dslx/frontend/ast.cc

```diff
@@ -420,6 +420,20 @@ absl::flat_hash_set<std::string> FreeVariables::Keys() const {
   return result;
 }
 
+std::string FreeVariables::ToString() const {
+  return absl::StrFormat(
+      "{%s}",
+      absl::StrJoin(values_, ", ", [](std::string* out, const auto& item) {
```

Comment:
Nit/optional: worth factoring out a named lambda for readability?

Reply:
Done.